Core/Src/main.c

```diff
@@ -65,6 +68,20 @@ void SystemClock_Config(void);
 int main(void)
 {
   /* USER CODE BEGIN 1 */
+  const uint16_t COUNTER_PERIOD = 60000;
+  const uint16_t ADC_MAX_VALUE = 1023;
+  const uint8_t TX_DATA[3] = {0x01, 0x0, 0x0};
+  const uint8_t RX_BIT_MASK = 0x03;
```

Comment:
This mask is correct but you need to apply it to rx[1] and not rx[0] right? according to that image i sent in my last review, the first byte that comes in (rx[0]) is all garbage data, the real data starts in second and third byte (rx[1] and rx[2]).

Reply:
Ah I assumed that the first byte to be received would be the first byte that contained actual data, but that makes no sense retrospectively...